### ISSUE TITLE
py-regex SHA-1 Collision

### ISSUE THREAD

**Issue by JusticeForMikeBrown** [LLNL/spack#5565]
Hi

I am trying to install `py-regex` with existing `python@2.7.13` however I get SHA-1 collision:

```
[root@node141 spack]# git log | head
commit 554937780bf9bad2e1e2b238aa95c6fd862b711e
Author: Massimiliano Culpo <massimiliano.culpo@googlemail.com>
Date:   Fri Sep 29 18:23:13 2017 +0200

    modules: specialized configure_options for external packages (#5543)
    
    closes #5473
    
    Prior to this PR we were not exiting early for external packages, which
    caused the `configure_options` property of the contexts to fail with

```

```
[root@node141 spack]# spack find -l -v python                                                                                                                                                                                                      
==> 9 installed packages.
-- linux-rhel6-x86_64 / gcc@4.4.6 -------------------------------
mpz42h4 python@2.7.13~tk~ucs4  ab5wsqm python@3.4.3~tk~ucs4  homuiqv python@3.6.0~tk~ucs4

-- linux-rhel6-x86_64 / gcc@6.3.0 -------------------------------
hccdc77 python@2.7.13+shared~tk~ucs4  5ogh6la python@2.7.13~tk~ucs4  t2ndbmk python@3.5.2+shared~tk~ucs4  pndebbt python@3.5.2~tk~ucs4  vov33zo python@3.6.0+shared~tk~ucs4  2pfngwu python@3.6.0~tk~ucs4
```

```
[root@node141 spack]# spack install py-regex %gcc@6.3.0 ^/hccdc77
==> Error: Specs python@2.7.13%gcc@6.3.0+pic+shared~tk~ucs4 arch=linux-rhel6-x86_64 ^bzip2@1.0.6%gcc@6.3.0+shared arch=linux-rhel6-x86_64 ^ncurses@6.0%gcc@6.3.0~symlinks arch=linux-rhel6-x86_64 ^openssl@1.0.2k%gcc@6.3.0 arch=linux-rhel6-x86_64 ^pkg-config@0.29.2%gcc@6.3.0+internal_glib arch=linux-rhel6-x86_64 ^readline@7.0%gcc@6.3.0 arch=linux-rhel6-x86_64 ^sqlite@3.18.0%gcc@6.3.0 arch=linux-rhel6-x86_64 ^zlib@1.2.11%gcc@6.3.0+pic+shared arch=linux-rhel6-x86_64 and python@2.7.13%gcc@6.3.0+shared~tk~ucs4 arch=linux-rhel6-x86_64 ^bzip2@1.0.6%gcc@6.3.0+shared arch=linux-rhel6-x86_64 ^ncurses@6.0%gcc@6.3.0~symlinks arch=linux-rhel6-x86_64 ^openssl@1.0.2k%gcc@6.3.0 arch=linux-rhel6-x86_64 ^readline@7.0%gcc@6.3.0 arch=linux-rhel6-x86_64 ^sqlite@3.18.0%gcc@6.3.0 arch=linux-rhel6-x86_64 ^zlib@1.2.11%gcc@6.3.0+pic+shared arch=linux-rhel6-x86_64 have the same SHA-1 prefix!
```
Concretization seems to work, though:

```
[root@node141 spack]# spack spec py-regex %gcc@6.3.0 ^/hccdc77
Input spec
--------------------------------
py-regex%gcc@6.3.0
    ^python@2.7.13%gcc@6.3.0+shared~tk~ucs4 arch=linux-rhel6-x86_64 
        ^bzip2@1.0.6%gcc@6.3.0+shared arch=linux-rhel6-x86_64 
        ^ncurses@6.0%gcc@6.3.0~symlinks arch=linux-rhel6-x86_64 
        ^openssl@1.0.2k%gcc@6.3.0 arch=linux-rhel6-x86_64 
            ^zlib@1.2.11%gcc@6.3.0+pic+shared arch=linux-rhel6-x86_64 
        ^readline@7.0%gcc@6.3.0 arch=linux-rhel6-x86_64 
        ^sqlite@3.18.0%gcc@6.3.0 arch=linux-rhel6-x86_64 

Normalized
--------------------------------
py-regex%gcc@6.3.0
    ^py-setuptools
        ^python@2.7.13%gcc@6.3.0+shared~tk~ucs4 arch=linux-rhel6-x86_64 
            ^bzip2@1.0.6%gcc@6.3.0+shared arch=linux-rhel6-x86_64 
            ^ncurses@6.0%gcc@6.3.0~symlinks arch=linux-rhel6-x86_64 
                ^pkg-config
            ^openssl@1.0.2k%gcc@6.3.0 arch=linux-rhel6-x86_64 
                ^zlib@1.2.11%gcc@6.3.0+pic+shared arch=linux-rhel6-x86_64 
            ^readline@7.0%gcc@6.3.0 arch=linux-rhel6-x86_64 
            ^sqlite@3.18.0%gcc@6.3.0 arch=linux-rhel6-x86_64 

Concretized
--------------------------------
py-regex@2017.07.11%gcc@6.3.0 arch=linux-rhel6-x86_64 
    ^py-setuptools@35.0.2%gcc@6.3.0 arch=linux-rhel6-x86_64 
        ^py-appdirs@1.4.3%gcc@6.3.0 arch=linux-rhel6-x86_64 
            ^python@2.7.13%gcc@6.3.0+pic+shared~tk~ucs4 arch=linux-rhel6-x86_64 
                ^bzip2@1.0.6%gcc@6.3.0+shared arch=linux-rhel6-x86_64 
                ^ncurses@6.0%gcc@6.3.0~symlinks arch=linux-rhel6-x86_64 
                    ^pkg-config@0.29.2%gcc@6.3.0+internal_glib arch=linux-rhel6-x86_64 
                ^openssl@1.0.2k%gcc@6.3.0 arch=linux-rhel6-x86_64 
                    ^zlib@1.2.11%gcc@6.3.0+pic+shared arch=linux-rhel6-x86_64 
                ^readline@7.0%gcc@6.3.0 arch=linux-rhel6-x86_64 
                ^sqlite@3.18.0%gcc@6.3.0 arch=linux-rhel6-x86_64 
        ^py-packaging@16.8%gcc@6.3.0 arch=linux-rhel6-x86_64 
            ^py-pyparsing@2.2.0%gcc@6.3.0 arch=linux-rhel6-x86_64 
            ^py-six@1.10.0%gcc@6.3.0 arch=linux-rhel6-x86_64 
```

**Comment by JusticeForMikeBrown** [comment]
I cannot install with another `python@2.7.13` `5ogh6la` either:

```
[root@node141 spack]# spack install py-regex %gcc@6.3.0 ^/5ogh6la
==> Error: Specs python@2.7.13%gcc@6.3.0+pic+shared~tk~ucs4 arch=linux-rhel6-x86_64 ^bzip2@1.0.6%gcc@6.3.0+shared arch=linux-rhel6-x86_64 ^ncurses@6.0%gcc@6.3.0~symlinks arch=linux-rhel6-x86_64 ^openssl@1.0.2k%gcc@6.3.0 arch=linux-rhel6-x86_64 ^pkg-config@0.29.2%gcc@6.3.0+internal_glib arch=linux-rhel6-x86_64 ^readline@7.0%gcc@6.3.0 arch=linux-rhel6-x86_64 ^sqlite@3.18.0%gcc@6.3.0 arch=linux-rhel6-x86_64 ^zlib@1.2.11%gcc@6.3.0+pic+shared arch=linux-rhel6-x86_64 and python@2.7.13%gcc@6.3.0~tk~ucs4 arch=linux-rhel6-x86_64 ^bzip2@1.0.6%gcc@6.3.0+shared arch=linux-rhel6-x86_64 ^ncurses@6.0%gcc@6.3.0~symlinks arch=linux-rhel6-x86_64 ^openssl@1.0.2k%gcc@6.3.0 arch=linux-rhel6-x86_64 ^readline@7.0%gcc@6.3.0 arch=linux-rhel6-x86_64 ^sqlite@3.18.0%gcc@6.3.0 arch=linux-rhel6-x86_64 ^zlib@1.2.11%gcc@6.3.0+pic+shared arch=linux-rhel6-x86_64 have the same SHA-1 prefix!

```

**Comment by scheibelp** [comment]
I can replicate your error and I'm working on a fix at #5580

**Comment by JusticeForMikeBrown** [comment]
Thanks @scheibelp!

**Comment by junghans** [comment]
Should be fixed by #5580, reopen if not.

**Comment by JusticeForMikeBrown** [comment]
I am reopening since puling the latest from develop broke Spack.  So I cannot yet verify.  See #5293.

**Comment by JusticeForMikeBrown** [comment]
@junghans why has my ability to reopen been removed?

**Comment by scheibelp** [comment]
#5293 is now resolved, and #5580 should fix this, so I am closing. Feel free to reopen if the issue persists for you.
